control_protocol.md

```diff
@@ -0,0 +1,293 @@
+Similarly, the Component shall store the Node ID and use it from this moment to generate its full ID.
+
+If a Component does send a message to someone without having signed in, the Coordinator shall refuse message handling and return an error.
+
+:::{mermaid}
+sequenceDiagram
+    Note over CA,Co1: ID "CA" is still free
+    CA ->> Co1: COORDINATOR|CA|SIGNIN
+    Note right of Co1: Connection identity "IA"
+    Note right of Co1: Stores "CA" with identity "IA"
+    Co1 ->> CA: Co1.CA|Co1.COORDINATOR|ACKNOWLEDGE: Node ID is "Co1"
+    Note left of CA: Stores "Co1" as Node ID
+    Note over CA,Co1: ID "CA" is already used
+    CA ->> Co1: COORDINATOR|CA|SIGNIN
+    Co1 ->> CA: CA|Co1.COORDINATOR|ERROR: ID "CA" is already used.
+    Note left of CA: May retry with another ID
+    Note over CA,Co1: "CA" has not send SIGNIN
+    Note left of CA: Wants to send a message to CB
+    CA ->> Co1: Co1.CB|CA|Content
+    Note right of Co1: Does not know CA
+    Co1 ->> CA: CA|Co1.COORDINATOR|ERROR:I do not know you
+    Note left of CA: Should send a SIGNIN message
+:::
+
+
+##### Heartbeat
+
+We use heartbeat to know, whether a communication partner is still online.
+
+Every message received counts as a heartbeat.
+
+A Component should and a Coordinator shall send a STATUS request and wait some time before considering a connection dead.
+A Coordinator shall follow the {ref}`sign out routine<sign-out>` for a signed in Component considered dead.
+
+:::{note}
+TBD: Respond to every non empty message with an empty one?
+:::
+
+
+(sign-out)=
+##### Signing out
+
+A Component should tell a Coordinator, when it stops working, with a SIGNOUT message.
+The Coordinator shall ACKNOWLEDGE the sign out and remove the ID from its address book.
+It shall also publish to the other Coordinators in the network that this Component signed out, see {ref}`Coordinator coordination<coordinator-coordination>`.
+
+:::{mermaid}
+sequenceDiagram
+    CA ->> Co1: COORDINATOR|Co1.CA|SIGNOUT
+    Co1 ->> CA: Co1.CA|Co1.COORDINATOR|ACKNOWLEDGE
+    Note right of Co1: Removes "CA" with identity "IA"<br> from address book
+    Note left of CA: Shall not send any message anymore
+:::
+
+
+#### Communication with other Components
+
+The following two examples show, how a message is transferred between two components `CA`, `CB` via one or two Coordinators.
+Coordinators shall send messages from their DEALER socket to other Coordinator's ROUTER socket.
+
+Coordinators shall hand on the message to the corresponding Coordinator or connected Component.
+
+
+:::{mermaid}
+sequenceDiagram
+    CA ->> Co1: Co1.CB|Co1.CA| Give me property A.
+    Co1 ->> CB: Co1.CB|Co1.CA| Give me property A.
+    Note left of CB: Reads property A
+    CB ->> Co1: Co1.CA|Co1.CB| Property A has value 5.
+    Co1 ->> CA: Co1.CA|Co1.CB| Property A has value 5.
+    Note over CA,Co1: The first message could be as well:
+    CA ->> Co1: CB|Co1.CA| Give me property A.
```

Comment:
Mermaid sequence diagrams can do[ "alt" ](https://mermaid.js.org/syntax/sequenceDiagram.html#alt)to incorporate alternative paths, so this could be included after the first line.

Reply:
Oh cool. Thanks

Reply:
Done